### PR TITLE
Feat/sent image

### PR DESCRIPTION
**Pull request by ZanBin**

### CODE REVIEW

File: example/src/Chat/Component/CustomInputMessage.tsx [hunk 1]
```diff
@@ -32,30 +37,27 @@ const CustomInputMessage: React.FC<ICustomInputMessage> = ({
...
-  //   //   if (DocumentPicker.isCancel(err)) {
-  //   //     // User cancelled the picker, exit any dialogs or menus and move on
-  //   //   } else {
-  //   //     throw err
-  //   //   }
-  //   // }
-  // };
+  const showImageLibrary = async () => {
+    try {
+      try {
```
Comment (loc-nguyenthien) [inline example/src/Chat/Component/CustomInputMessage.tsx:42]: double `try-catch`?

**Review by loc-nguyenthien — CHANGES_REQUESTED** [review]

File: src/Chat/ChatProvider.tsx [hunk 2]
```diff
@@ -96,34 +92,13 @@ export const ChatProvider = React.forwardRef<any, ChatScreenProps>(
         GiftedChat.append(previousMessages, [messages])
       );
 
-      let file;
```
Comment (baonguyenhsts) [inline src/Chat/ChatProvider.tsx:99]: why do you remove this line?

File: src/Chat/ChatProvider.tsx [hunk 3]
```diff
@@ -96,34 +92,13 @@ export const ChatProvider = React.forwardRef<any, ChatScreenProps>(
...
-      //           // fileUrl: fileUrl,
-      //           // fileName: messages?.fileName,
-      //           // fileSize: messages?.fileSize,
-      //           extension: messages?.extension,
-      //         };
+      let file = {
+        imageUrl: messages?.imageUrl,
+        extension: messages?.extension,
+        type: messageType
+      };
```
Comment (baonguyenhsts) [inline src/Chat/ChatProvider.tsx:100]: What happen if the file is
```
{
        imageUrl: undefined
        extension: undefined,
        type: undefined
      }
```

File: src/Services/Firestore/firestore.ts [hunk 4]
```diff
@@ -37,7 +37,7 @@ export class FirestoreServices {
     | FirebaseFirestoreTypes.QueryDocumentSnapshot<MessageProps>
     | undefined;
 
-  constructor() {}
+  constructor() { }
```
Comment (baonguyenhsts) [inline src/Services/Firestore/firestore.ts:40]: prettier config

File: src/Services/Firestore/firestore.ts [hunk 5]
```diff
@@ -62,80 +62,67 @@ export class FirestoreServices {
...
-      status: 'pending',
-      senderId: this.userId,
-      created: created,
-      text: message,
-      ...file,
-    };
-    await this.updateLatestMessageInChannel(message);
+    try {
+      if (!this.conversationId) {
+        return;
```
Comment (baonguyenhsts) [inline src/Services/Firestore/firestore.ts:67]: Should log error if the conversationId doesn't exist

File: src/Services/Firestore/firestore.ts [hunk 6]
```diff
@@ -62,80 +62,67 @@ export class FirestoreServices {
...
+      }
+      let message = text;
+      const messageData = {
+        readBy: {},
+        status: 'pending',
```
Comment (baonguyenhsts) [inline src/Services/Firestore/firestore.ts:72]: use MESSAGE_STATUS constant

File: src/Services/Firestore/firestore.ts [hunk 7]
```diff
@@ -62,80 +62,67 @@ export class FirestoreServices {
...
+        await this.updateLatestMessageInChannel(message);
+        firestore()
+          .collection<MessageProps>(
+            `${FireStoreCollection.conversations}/${this.conversationId}/${FireStoreCollection.messages}`
+          )
+          .add(messageData)
+          .then((snapShot) => {
+            snapShot
+              .update({
+                status: 'sent',
```
Comment (baonguyenhsts) [inline src/Services/Firestore/firestore.ts:93]: use MESSAGE_STATUS constant

File: src/Services/Firestore/firestore.ts [hunk 8]
```diff
@@ -62,80 +62,67 @@ export class FirestoreServices {
...
+      if (file) {
+        uploadFileToFirebase(
+          file.imageUrl,
+          file.extension,
+          this.conversationId
+        ).then((res) => {
+          storage()
+            .ref(res.metadata.fullPath)
+            .getDownloadURL()
+            .then((imageUrl) => {
```
Comment (baonguyenhsts) [inline src/Services/Firestore/firestore.ts:109]:
```suggestion
            .then((downloadURL) => {
```

File: src/Services/Firestore/firestore.ts [hunk 9]
```diff
@@ -62,80 +62,67 @@ export class FirestoreServices {
...
+              firestore()
+                .collection<MessageProps>(
+                  `${FireStoreCollection.conversations}/${this.conversationId}/${FireStoreCollection.messages}`
+                )
+                .add(messageData)
+                .then((snapShot) => {
+                  snapShot.update({
+                    imageUrl,
+                    status: 'sent',
```
Comment (baonguyenhsts) [inline src/Services/Firestore/firestore.ts:118]: use MESSAGE_STATUS constant

File: src/Services/Firestore/firestore.ts [hunk 10]
```diff
@@ -414,8 +401,31 @@ export class FirestoreServices {
...
-    return { ...conversationData, id: conversationRef.id };
+    return {...conversationData, id: conversationRef.id};
+  };
+
+  getConservation = async (userId: string, memberId: string) => {
+    let conversationId = '';
+    await firestore()
+      .collection(`${FireStoreCollection.conversations}`)
+      .get()
+      .then((e) => {
```
Comment (baonguyenhsts) [inline src/Services/Firestore/firestore.ts:413]: Why do you use `async/await` with `then`?

**Review by baonguyenhsts — CHANGES_REQUESTED** [review]

File: example/src/Chat/Component/CustomInputMessage.tsx [hunk 11]
```diff
@@ -5,18 +5,23 @@ import React, {useState} from 'react';
...
 import {Composer, InputToolbarProps, SendProps} from 'react-native-gifted-chat';
 import {PressAbleIcon} from '../../Components';
+import ImagePicker from 'react-native-image-picker';
 
 interface ICustomInputMessage extends InputToolbarProps<any>, SendProps<any> {
   isShowPhotoGallery: boolean;
   togglePhotoGallery: (value: boolean) => void;
 }
 
+const byteToMB = 1048576;
```
Comment (hoangSTS) [inline example/src/Chat/Component/CustomInputMessage.tsx:15]: what does it mean and used for?

File: example/src/Chat/Component/CustomInputMessage.tsx [hunk 12]
```diff
@@ -32,30 +37,26 @@ const CustomInputMessage: React.FC<ICustomInputMessage> = ({
...
+          mediaType: 'photo',
+          includeBase64: false,
+          includeExtra: true,
+          quality: 1,
+        },
+        (res: ImagePicker.ImagePickerResponse) => {
+          const {assets} = res;
+          if (assets && assets?.length > 0) {
+            const {type, uri} = assets[0];
+            onSend?.({imageUrl: uri, extension: type, type: 'image'}, true);
```
Comment (hoangSTS) [inline example/src/Chat/Component/CustomInputMessage.tsx:53]: replace 'image' by IMAGE_TYPE above

**Pull request closed without merging**